### DISCUSSION TITLE
Is there a way to get Auto Invoked function results back from the LLM ChatMessageContent object?

### DISCUSSION THREAD

**Discussion by preezie-yasir** [microsoft/semantic-kernel#5926] [Q&A] [answered]
![image](https://github.com/microsoft/semantic-kernel/assets/115682529/551a02d2-1414-4e50-99a8-fdcabd974888)
I have line 92 sending a user prompt to the kernels chat completion api. This may auto invoke functions which return results i.e searching a database for a product and returning that JSON object.

In between line 92 and 97, i want to get back the results from the functions that were executed by the LLM but i cant seem to find a way to do so. Any assistance is appreciated.

**Comment by dmytrostruk** [comment, marked as answer]
@preezie-yasir Since you have access to `chatHistory` object, you can get it from there. We add each function result to `chatHistory` after automatic function invocation. Let me know if that works for you.

**Reply by preezie-yasir** [reply]
Yep works perfectly. Completely forgot to check the chat history object!
